orchagent/aclorch.cpp

```diff
@@ -3168,6 +3227,30 @@ void AclOrch::initDefaultTableTypes()
             .build()
     );
 
+
+    addAclTableType(
+        builder.withName(TABLE_TYPE_L3V4V6)
+            .withBindPointType(SAI_ACL_BIND_POINT_TYPE_PORT)
+            .withBindPointType(SAI_ACL_BIND_POINT_TYPE_LAG)
+            .withMatch(make_shared<AclTableMatch>(SAI_ACL_TABLE_ATTR_FIELD_ETHER_TYPE))
+            .withMatch(make_shared<AclTableMatch>(SAI_ACL_TABLE_ATTR_FIELD_OUTER_VLAN_ID))
+            .withMatch(make_shared<AclTableMatch>(SAI_ACL_TABLE_ATTR_FIELD_ACL_IP_TYPE))
+            .withMatch(make_shared<AclTableMatch>(SAI_ACL_TABLE_ATTR_FIELD_SRC_IP))
+            .withMatch(make_shared<AclTableMatch>(SAI_ACL_TABLE_ATTR_FIELD_DST_IP))
+            .withMatch(make_shared<AclTableMatch>(SAI_ACL_TABLE_ATTR_FIELD_ICMP_TYPE))
+            .withMatch(make_shared<AclTableMatch>(SAI_ACL_TABLE_ATTR_FIELD_ICMP_CODE))
+            .withMatch(make_shared<AclTableMatch>(SAI_ACL_TABLE_ATTR_FIELD_IP_PROTOCOL))
+            .withMatch(make_shared<AclTableMatch>(SAI_ACL_TABLE_ATTR_FIELD_SRC_IPV6))
+            .withMatch(make_shared<AclTableMatch>(SAI_ACL_TABLE_ATTR_FIELD_DST_IPV6))
+            .withMatch(make_shared<AclTableMatch>(SAI_ACL_TABLE_ATTR_FIELD_ICMPV6_CODE))
+            .withMatch(make_shared<AclTableMatch>(SAI_ACL_TABLE_ATTR_FIELD_ICMPV6_TYPE))
+            .withMatch(make_shared<AclTableMatch>(SAI_ACL_TABLE_ATTR_FIELD_IPV6_NEXT_HEADER))
+            .withMatch(make_shared<AclTableMatch>(SAI_ACL_TABLE_ATTR_FIELD_L4_SRC_PORT))
+            .withMatch(make_shared<AclTableMatch>(SAI_ACL_TABLE_ATTR_FIELD_L4_DST_PORT))
+            .withMatch(make_shared<AclTableMatch>(SAI_ACL_TABLE_ATTR_FIELD_TCP_FLAGS))
+            .build()
+    );
```

Comment:
Two questions
1. Do we need to handle conflicts? For example, one ACL rule contains both "SAI_ACL_TABLE_ATTR_FIELD_SRC_IP` and `SAI_ACL_TABLE_ATTR_FIELD_SRC_IPV6`.
2. If a rule doesn't contain  `SAI_ACL_TABLE_ATTR_FIELD_ETHER_TYPE` or `SAI_ACL_TABLE_ATTR_FIELD_ACL_IP_TYPE`, should it be allowed ?

Reply:
Thanks, we need to ensure the TCAM slice is wide enough to hold so many fields.

Reply:
> Two questions
> 
> 1.  Do we need to handle conflicts? For example, one ACL rule contains both `SAI_ACL_TABLE_ATTR_FIELD_SRC_IP` and `SAI_ACL_TABLE_ATTR_FIELD_SRC_IPV6`
> 2.  If a rule doesn't contain `SAI_ACL_TABLE_ATTR_FIELD_ETHER_TYPE` or `SAI_ACL_TABLE_ATTR_FIELD_ACL_IP_TYPE`, should it be allowed ?

1.  IMO, orchagent should not handle these conflicts-- similar to how orchagent does not handle one ACL rule containing both ICMP type and L4 source port.
2.  It should be allowed by orchagent for reasons:
    1.  Allow users to specify ACL rules that match only IN\_PORTS
    2.  Based on [feedback](https://github.com/sonic-net/SONiC/pull/1267#discussion_r1174595677) from other SAI vendors, we updated the HLD to specify that when IP\_type or Ether-type is not specified, the underlying SAI platform to handle based on their capability
        1.  install a single rule if the hardware supports it, or
        2.  make multiple rules (one rule per IP type), or
        3.  throw error as unsupported.

Reply:
Regarding #1, If orchagent doesn't handle this exception, then SAI has to handle it.  But I don't think we have clear agreement about how SAI handles this.

Reply:
Added the checks and unit tests.

test\_acl\_l3v4v6.py::TestAcl::test\_ValidAclRuleCreation\_sip\_dip PASSED                                                                          test\_acl\_l3v4v6.py::TestAcl::test\_InvalidAclRuleCreation\_sip\_sipv6 PASSED                                                                    test\_acl\_l3v4v6.py::TestAcl::test\_InvalidAclRuleCreation\_dip\_sipv6 PASSED